### DISCUSSION TITLE
Application lifecycle management and multi module maven support

### DISCUSSION THREAD

**Discussion by moazharuddin** [micronaut-projects/micronaut-core#5848] [Q&A] [answered]
I am currently using micronaut 2.5.5 as im new to micronaut i need to understand the application lifecycle, as how it starts the application without any annotation like spring @springbootapplication. Also i want to know if there is any support for multi module maven project like spring?

**Comment by ilopmar** [comment, marked as answer]
We don't need an annotation to start Micronaut. See https://micronaut.io/launch?javaVersion=JDK_11&lang=JAVA&build=MAVEN&test=JUNIT&name=demo&package=com.example&type=DEFAULT&version=2.5.9&activity=preview&showing=/src/main/java/com/example/Application.java

Regarding the multi-module support, yes that is supported for both Maven and Gradle, but it has nothing to do with Micronaut. You need to configure the multi-project yourself.

**Reply by moazharuddin** [reply]
Thanks for your reply, but i was looking deep on how the applicationcontext run get enabled and attached to the embedded server. I got it myself.. May be clear documentation would help. thanks..

**Reply by ilopmar** [reply]
If you find something that is not clear enough or that can be improved, please feel free to send a PR to update the docs. You only need to click on the "Improve this docs" button on the right side of the documentation: https://docs.micronaut.io/latest/guide/index.html
